modules/models_settings.py

```diff
@@ -48,6 +49,10 @@ def get_model_metadata(model):
         metadata = metadata_gguf.load_metadata(model_file)
         if 'llama.context_length' in metadata:
             model_settings['n_ctx'] = metadata['llama.context_length']
+        if 'llama.rope.scale_linear' in metadata:
+            model_settings['compress_pos_emb'] = metadata['llama.rope.scale_linear']
```

Comment:
Should this be `1 / metadata['llama.rope.scale_linear']`? iirc llama.cpp uses fractional values like 0.5 and 0.25, while the UI expects integer values like 2 and 4.

Reply:
@oobabooga no, it represents an int, not a float.
So if the model is 16k, from the metadata we get `llama.rope.scale_linear = 4`